lint_test/avoid_final_with_getter_test.dart

```diff
@@ -0,0 +1,27 @@
+// ignore_for_file: type_annotate_public_apis, prefer_match_file_name, unused_local_variable
+
+/// Check final private field with getter fail
+/// `avoid_final_with_getter`
+
+class Fail {
+  // expect_lint: avoid_final_with_getter
+  final int _myField = 0;
+
+  int get myField => _myField;
+}
+
+class Skipped {
+  final int _myField = 0;
+
+  int get myField => _myField + 1; // it is not a getter for the field
+}
+
+class Skipped2 {
+  final int _myField = 0;
+
+  int get myFieldInt => _myField; // it is not a getter for the field too
```

Comment:
I think in this case it should report the lint since it is a getter for the field.

Reply:
yes, but I think it is necessary to additionally check whether there is another getter or variable with the same name.

Reply:
I don't think it make sense to search for a getter with exactly the same name as a variable. If we have a private variable and a public getter(with any name) that returns this variable we should show lint. Also, I think it makes sense to highlight the getter, but not the variable, so the user can see why we are saying that this field should be just public.